src/spell/modules/spell/spell/src/kv/primitive.rs

```diff
@@ -22,21 +27,34 @@ pub fn set_string(key: &str, value: String) -> UnitValue {
 
 pub fn read_string(statement: &mut Statement, idx: usize) -> eyre::Result<Option<String>> {
     if let State::Row = statement.next()? {
-        Ok(Some(statement.read::<String>(idx)?))
+        let read_value = statement.read::<String>(idx)?;
+        // Need to clone because otherwise `Some(read_value)` morphs into `None` O.o
+        Ok(Some(read_value.to_string()))
     } else {
         Ok(None)
     }
+
 }
 
 #[marine]
 pub fn get_string(key: &str) -> StringValue {
     let result: eyre::Result<Option<String>> = try {
         let conn = db();
-        let mut statement = conn.prepare("SELECT string FROM kv WHERE key = ?")?;
+        // As long as an empty string is saved a NULL value, we can determine that the value is a string
```

Comment:
We could also have a column `type` in db, and write `string | u32 | list` there. Then it might be easier, I guess.